docs/decisions/0003-migrating-to-react-virtuoso.md

```diff
@@ -0,0 +1,31 @@
+# Migrating Parsley Virtualization to react-virtuoso
+
+* status: proposed
+* date: 2022-02-22
+* authors: Mohamed Khelif
+
+## Context and Problem Statement
+
+Parsley currently uses the [react-virtualized](https://github.com/bvaughn/react-virtualized) library to render log files. This library was the only library we could find that can support a continuous virtualized view capable of rendering over 800k lines, which is often the case with some log files. Unfortunately it has many problems, some of which include:
+
+* It is no longer maintained by its owner.
+* It is unable to maintain a scroll position when scrolling to previously unvisited lines due to some buggy scroll position compensation logic. (Which ironically is probably the thing that lets it render so many lines). This causes the log file to "jump" dramatically.
+  * It needs to maintain a cache of the sizes of each log line.
```

Comment:
indentation?

FYI there's a build-in markdown preview mode in vs code: `cmd-k, v`

Reply:
This is intentional it is meant to be a sub-point. It reflects the contents of the ticket. https://jira.mongodb.org/browse/EVG-18993

Reply:
Ah gotcha, sorry!